cmd/dlv/cmds/commands.go

```diff
@@ -272,36 +271,46 @@ https://github.com/mozilla/rr
 
 func debugCmd(cmd *cobra.Command, args []string) {
 	status := func() int {
-		var pkg string
-		dlvArgs, targetArgs := splitArgs(cmd, args)
-
-		if len(dlvArgs) > 0 {
-			pkg = args[0]
+		debugname := cmd.Flag("output").Value.String()
+		if !strings.HasPrefix(debugname, "/") {
+			debugname = "./" + debugname
 		}
-		err := gobuild(debugname, pkg)
+		debugname, err := filepath.Abs(debugname)
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "%v\n", err)
 			return 1
 		}
-		fp, err := filepath.Abs("./" + debugname)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "%v\n", err)
-			return 1
+
+		var pkg string
+		dlvArgs, targetArgs := splitArgs(cmd, args)
+
+		if len(dlvArgs) > 0 {
+			pkg = args[0]
 		}
-		defer os.Remove(fp)
-		abs, err := filepath.Abs(debugname)
+		err = gobuild(debugname, pkg)
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "%v\n", err)
 			return 1
 		}
-		processArgs := append([]string{abs}, targetArgs...)
+		defer os.Remove(debugname)
+		processArgs := append([]string{debugname}, targetArgs...)
 		return execute(0, processArgs, conf, "", executingGeneratedFile)
 	}()
 	os.Exit(status)
 }
 
 func traceCmd(cmd *cobra.Command, args []string) {
 	status := func() int {
+		debugname := cmd.Flag("output").Value.String()
+		if !strings.HasPrefix(debugname, "/") {
+			debugname = "./" + debugname
+		}
+		debugname, err := filepath.Abs(debugname)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			return 1
+		}
+
```

Comment:
This block of code is the same in debug, test and trace. I think it should be its own function.

Reply:
Actually the extra `filepath.IsAbs()` and such isn't needed, and can just be reduces to `filepath.Abs()` plus error check – I think it's okay to keep this inline like this?